data_structures/queue_using_two_stacks.cpp

```diff
@@ -0,0 +1,144 @@
+/**
+ * @author [shoniavika](https://github.com/shoniavika)
+ * @file
+ *
+ * Implementation of a Queue using two Stacks.
+ */
+
+#include <cassert>
+#include <iostream>
+
+#include "./stack.h"
+
+/**
+ * Queue data structure. Stores elements in FIFO
+ * (first-in-first-out) manner.
+ *     methods:
+ *         push(int x)  :  Pushes x to the back of queue.
+ *         pop()        :  Removes an element from the front
+ *         peek()       :  Returns first element, without removing
+ *         empty()      :  Returns whether the queue is empty
+ */
+class MyQueue {
+ private:
+    stack<int> s1, s2;
+
+    /**
+     * Helper function for rearranging elements in stacks
+     */
+    void rearrange();
+
+ public:
+    /**
+     * Constructor for queue.
+     */
+    MyQueue() = default;
+
+    /**
+     * Pushes x to the back of queue.
+     */
+    void push(int x);
+
+    /**
+     * Removes an element from the front of the queue.
+     */
+    int pop();
```

Comment:
```suggestion
    T pop();
```

Reply:
Thanks for your comments, these were really helpful, learned a lot. I corrected all nuances. However, I neither think that pop method should be const nor should it return by reference. Let me know if it's otherwise!

Reply:
It should be constant because then, people cannot perform
`tets_queue.pop() = 3`
This type of assignment would be meaningless when not using a reference output but would be possible if `const` is not used.